enhancements/api-review/custom-tags-aws.md

```diff
@@ -73,71 +95,373 @@ Note existing unchanged behavior: The installer will apply these tags to all AWS
+  This can be supported by allowing edit on mutable `.spec.platformSpec.aws.resourceTags` field in Infrastructure CR.
+
+- As a cluster administrator of OpenShift, I would like to be able to remove user-defined tags from `.spec.platformSpec.aws.resourceTags` without deleting for AWS resources.
+  This enables, user-defined tags to be modified on AWS without being overridden by OpenShift operators. The user can adopt new 3rd party component to manage tags.
+
+- As a cluster administrator of OpenShift, I expect user-defined tags added in Infrastructure CR are reconciled and desired user-defined tags maintained on AWS resources.
+  The latest updates on `.spec.platformSpec.aws.resourceTags` field must be reconciled to AWS resources and desired user-defined tags maintained when being managed by OpenShift.
+  Any modifications which are external to OpenShift must be reconciled to desired user-defined tags from `.spec.platformSpec.aws.resourceTags`.
+
+The following user stories are added to support existing methods of updating user-defined tags which override `.spec.platformSpec.aws.resourceTags` to avoid breaking change.
+As the existing cluster might be configured with user-defined tags from kube resources, the precedence is given to user-defined tags in kube resources.
+- As a cluster administrator of OpenShift, I expect user-defined tags added in kube resources override the entries in `.spec.platformSpec.aws.resourceTags`.
+
+- As a cluster administrator of OpenShift, I expect user-defined tags added in kube resources must not be overridden by update action or by removing user-defined tags from `.spec.platformSpec.aws.resourceTags`.
+
+### API Extensions
+
+This proposal edits `Infrastructure.config.openshift.io/v1` type
+
+The Infrastructure resource example to use spec for api changes
+
+```go
+// AWSPlatformSpec holds the desired state of the Amazon Web Services Infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type AWSPlatformSpec struct {
+    // Existing fields
+    ...
+    // ResourceTags is a list of additional tags to apply to AWS resources created for the cluster.
+    // See https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html for information on tagging AWS resources.
+    // AWS supports a maximum of 10 tags per resource. OpenShift reserves 5 tags for its use, leaving 5 tags
+    // available for the user.
+    // ResourceTags field is mutable and items can be removed.
+    // When a tag is removed from ResourceTags, the tag still persists on AWS resources.
+    // +kubebuilder:validation:MaxItems=10
+    // +optional
+    ResourceTags []AWSResourceTag `json:"resourceTags,omitempty"`
+}
+
+// Existing type
+// AWSResourceTag is a tag to apply to AWS resources created for the cluster.
+type AWSResourceTag struct {
```

Comment:
On the call, we discussed using a new type so that, in the future, if the spec and status need to drift, we don't add extra fields to the status that aren't required.
```suggestion
// AWSResourceTagSpec is a tag to apply to AWS resources created for the cluster.
type AWSResourceTagSpec struct {
```